### ISSUE TITLE
Code replication issue

### ISSUE THREAD

**Issue by Nighttell** [James-QiuHaoran/LLM-serving-with-proxy-models#10]
Hello, I tried to reproduce your code, but I couldn't get the results in your paper.

My only change was batch size, and I ran the following code directly，

#predictor training all LLMs
python preprocess_dataset.py --task_type 0 --data_size 10 --all_models
python latency_prediction.py --task_type 0 --data_size 10 --all_models

But the result is very exaggerated：

Validation metrics after training:
L1 error: 121.2489
MSE: 27016.6429
Start inference...
Saved results to ./results/predictions_all_models_warmup_reg_mse_10K.csv

Then I tried several other tasks and found that the results were not very satisfactory.
Please tell me where I went wrong?

Looking forward to your answer！
Thanks！

**Comment by James-QiuHaoran** [comment]
Please try with a larger `data_size`. In your case, `--data_size 10` means that BERT training dataset is only 10K, which is not enough for the predictor to reach a sufficient prediction accuracy. In the paper, the actual training dataset size is the entire LMSYS-Chat dataset.

**Comment by Nighttell** [comment]
thanks for your reply!
